proofs/alf/cvc5/Cvc5.smt3

```diff
@@ -128,33 +128,63 @@
   :conclusion (= t (run_evaluate t))
 )
 
-(program $is_poly_norm ((T Type) (U Type) (a T) (b T) (a1 U) (a2 U) (b1 U) (b2 U))
-  (T T) Bool
-  (
-    (($is_poly_norm (> a1 a2) (> b1 b2))   ($is_poly_norm_arith_mod_c (- a1 a2) (- b1 b2) true))
-    (($is_poly_norm (>= a1 a2) (>= b1 b2)) ($is_poly_norm_arith_mod_c (- a1 a2) (- b1 b2) true))
-    (($is_poly_norm (<= a1 a2) (<= b1 b2)) ($is_poly_norm_arith_mod_c (- a1 a2) (- b1 b2) true))
-    (($is_poly_norm (< a1 a2) (< b1 b2))   ($is_poly_norm_arith_mod_c (- a1 a2) (- b1 b2) true))
-    (($is_poly_norm (= a1 a2) (= b1 b2))   (alf.ite (is_arith_type (alf.typeof a1)) 
-                                           (alf.ite (is_arith_type (alf.typeof b1))
-                                             ($is_poly_norm_arith_mod_c (- a1 a2) (- b1 b2) false)
-                                             false)
-                                           false)) ; TODO: bv
-    (($is_poly_norm a b)                   (alf.ite (is_arith_type (alf.typeof a)) 
-                                           (alf.ite (is_arith_type (alf.typeof b))
-                                             ($is_poly_norm_arith a b)
-                                             false)
-                                           false)) ; TODO: bv
-  )
-)
-
-; ProofRule::ARITH_POLY_NORM
+; rule: arith_poly_norm
+; implements: ProofRule::ARITH_POLY_NORM
+; args:
+; - a U: The left-hand side of the equality.
+; - b U: The right-hand side of the equality.
+; requires: >
+;   That the arithmetic/bitvector expressions a and b are equal up polynomial
+;   normalization.
+; conclusion: The given equality.
 (declare-rule arith_poly_norm ((U Type) (a U) (b U))
   :args ((= a b))
   :requires ((($is_poly_norm a b) true))
   :conclusion (= a b)
 )
 
+; define: $is_poly_norm_rel
+; args:
+; - r (-> U U Bool): The binary relation to check.
+; - cx U: The scaling factor for (- x1 x2).
+; - cy U: The scaling factor for (- y1 y2).
+; return: >
+;   The result of checking if r is a polynomial normalization relation. If r is
+;   any of '<', '<=', '>=', '>', the scaling factors must have the same sign.
+;   For bitvectors, they must be set to 1. Returns false for any other relation.
```

Comment:
This is not yet implemented for bitvectors though, right? Can you make sure the doc is in sync?

Reply:
This is not an issue with the implementation of this rule or cvc5. It's an issue with the implementation of `arith_poly_norm` in ALF. I updated `arith_poly_norm`'s documentation to make that clear.